flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/FileExecutionGraphInfoStoreTest.java

```diff
@@ -341,11 +346,29 @@ public void testMaximumCapacity() throws IOException {
     /** Tests that a session cluster can terminate gracefully when jobs are still running. */
     @Test
     public void testPutSuspendedJobOnClusterShutdown() throws Exception {
+        final Duration timeout = Duration.ofSeconds(5);
         try (final MiniCluster miniCluster =
                 new PersistingMiniCluster(new MiniClusterConfiguration.Builder().build())) {
             miniCluster.start();
             final JobGraph jobGraph = JobGraphTestUtils.singleNoOpJobGraph();
+            final JobID jobId = jobGraph.getJobID();
             miniCluster.submitJob(jobGraph);
+            CommonTestUtils.waitUntilCondition(
+                    () -> {
+                        // Wait until job is FINISHED which indicates that the resource manager was
+                        // reachable
+                        try {
+                            return miniCluster.getJobStatus(jobId).get() == JobStatus.FINISHED;
```

Comment:
Doesn't that invalidate the test?
> Tests that a session cluster can terminate gracefully when jobs are still running.

Reply:
(although it's not like the test was actually verifying this before...)

Reply:
You are completely right even the first version was not 100% ensuring the behavior. I changed the test setup to use blocking job vertex and wait until it is running before doing the shutdown.